sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -445,14 +445,13 @@ export class ManagementClient extends LinkEntity<RequestResponseLink> {
     throwTypeErrorIfParameterNotLong(connId, "fromSequenceNumber", fromSequenceNumber);
 
     // Checks for maxMessageCount
-    if (maxMessageCount !== undefined) {
-      throwTypeErrorIfParameterTypeMismatch(connId, "maxMessageCount", maxMessageCount, "number");
-      if (maxMessageCount <= 0) {
-        return [];
-      }
-    } else {
+    if (maxMessageCount == undefined) {
       maxMessageCount = 1;
     }
+    maxMessageCount = Number(maxMessageCount);
+    if (isNaN(maxMessageCount) || maxMessageCount < 1) {
```

Comment:
Do we care about protecting against `Infinity` and `-Infinity`? If so, `isFinite` covers everything `isNaN` does, plus those cases.

(Side note: interestingly `isNaN(null)` returns false, and `isFinite(null)` returns true so both are weird in that regards, but `null < 1` returns true so we're covered!)

Reply:
There is a check a few lines above that takes care of null and undefined, but on seconds thoughts, I feel that it should handle only undefined i.e. the case where user passes nothing (default to 1) vs case where user passes something and that is null. Should we care about that? Or maybe in both null and undefined, we should throw saying that a number is expected?

I am considering just the below now...

```js
if (typeof maxMessageCount !== "number" && maxMessageCount < 1) {
   // throw error
}
```

Reply:
Ah, so don't let users pass in strings instead of numbers? I'm not opposed to that (though just FYI, NaN would pass that check :smile:)

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-js/pull/12453/commits/e45d0644a346d7940c78bfc6d2a5d2acec77233a